# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/src/spatial_node.rs
# Commit: 431c5142ff3a
# Full Hash: 431c5142ff3ad102328d90cb29470b20ccd1271e
# Author: Glenn Watson <github@intuitionlibrary.com>
# Date: 2019-06-13 21:50:57
# Regressor Bug: 1558106
# File Overlap Count: 1
# Description:
#   Bug 1558106 - Support picture caching for multiple scroll roots. r=kvark
#   
#   This patch implements the majority of the planned picture caching
#   improvements. It supports most of the functionality required to
#   (as a follow up) support OS compositor integration. It also improves
# ==============================================================================

diff -r 855d557b82a2 -r 431c5142ff3a gfx/wr/webrender/src/spatial_node.rs
--- a/gfx/wr/webrender/src/spatial_node.rs	Wed Jun 12 16:18:22 2019 +0000
+++ b/gfx/wr/webrender/src/spatial_node.rs	Thu Jun 13 04:43:56 2019 +0000
@@ -362,7 +362,8 @@
                                 cur_transform.flatten_z_output();
                             }
                             let world_transform = cur_transform.post_mul(&parent_system.world_transform);
-                            info.invertible = world_transform.determinant() != 0.0;
+                            let determinant = world_transform.determinant();
+                            info.invertible = determinant != 0.0 && !determinant.is_nan();
 
                             CoordinateSystem {
                                 transform,